include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1037 @@
+
+            // end kernel if we have no frames within the supercell
+            if( !frame.isValid( ) )
+                return;
+
+            auto accFilter = filter(
+                acc,
+                superCellIdx - mapper.getGuardingSuperCells( ),
+                WorkerCfg< numWorkers >{ workerIdx }
+            );
+
+            memory::CtxArray<
+                typename FramePtr::type::ParticleType,
+                ParticleDomCfg
+            >
+            currentParticleCtx(
+                workerIdx,
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    auto particle = frame[ linearIdx ];
+                    /* - only particles from the last frame must be checked
+                     * - all other particles are always valid
+                     */
+                    if( particle[ multiMask_ ] != 1 )
+                        particle.setHandleInvalid( );
+                    return particle;
+                }
+            );
+
+            while( frame.isValid( ) )
+            {
+                // loop over all particles in the frame
+                ForEachIdx< ParticleDomCfg > forEachParticle( workerIdx );
+
+                forEachParticle(
+                    [&](
+                        uint32_t const linearIdx,
+                        uint32_t const idx
+                    )
+                    {
+                        /* get one particle */
+                        auto & particle = currentParticleCtx[ idx ];
+                        if( accFilter( acc, particle ) ){
+
+                            float_X const weighting = particle[ weighting_ ];
+                            float3_X const mom = particle[ momentum_ ] / weighting;
+                            floatD_X const pos = particle[ position_ ];
+
+                            lcellId_t const cellIdx = particle[ localCellIdx_ ];
+                            DataSpace< simDim > const frameCellOffset(
+                                DataSpaceOperations< simDim >::template
+                                map< MappingDesc::SuperCellSize > ( cellIdx )
+                            );
+                            auto const localSupercellStart = (
+                                superCellIdx -
+                                mapper.getGuardingSuperCells()
+                            ) * MappingDesc::SuperCellSize::toRT();
+                            int const index_y = frameCellOffset.y();
+                            auto const globalCellOffset = globalOffset
+                                                        + localSupercellStart
+                                                        + frameCellOffset;
+                            float_X const posX = ( float_X( globalCellOffset.x() ) + pos.x() ) * cellSize.x();
+
+                            atomicAdd( &(shCount_e[index_y]), weighting, ::alpaka::hierarchy::Threads{} );
```

Comment:
This should be normalized before you accumulate the value like  https://github.com/ComputationalRadiationPhysics/picongpu/blob/f6f8ebd7a8771a1ed7d07e6f14d03011668235c0/include/picongpu/plugins/BinEnergyParticles.hpp#L242-L256

The problem is that you get very fast an float overflow in astro simulations.